src/Tasks/dependency-check-build-task/dependency-check-build-task.ts

```diff
@@ -212,8 +285,31 @@ async function getZipUrl(version: string): Promise<void> {
 async function unzipFromUrl(zipUrl: string, unzipLocation: string): Promise<void> {
     let fileName = path.basename(url.parse(zipUrl).pathname);
     let zipLocation = tl.resolve(fileName)
+    let tmpError = null;
+    let response = null;
+    let downloadErrorRetries = 5;
+    
+    do {
+        tmpError = null;
+        try {
+            await console.log('Downloading ZIP from "' + zipUrl + '"...');
+            response = await client.get(zipUrl);
+            await logDebug('done downloading');
+        }
+        catch(error) {
+            tmpError = error;
+            downloadErrorRetries--;
+            await console.error('Error trying to download ZIP (' + (downloadErrorRetries+1) + ' tries left)');
+            await console.error(error);
+        }
+    }
+    while(tmpError !== null && downloadErrorRetries >= 0);
+    
+    if(tmpError !== null) {
```

Comment:
If even after all those retries we still had an error downloading the ZIP then throw the last generated error.